diff --git a/external/linphone-sdk/external/opus/CMakeLists.txt b/external/linphone-sdk/external/opus/CMakeLists.txt
index fcf034b1..d766e47d 100644
--- a/external/linphone-sdk/external/opus/CMakeLists.txt
+++ b/external/linphone-sdk/external/opus/CMakeLists.txt
@@ -299,7 +299,9 @@ set_target_properties(opus
                                  VERSION
                                  ${OPUS_LIBRARY_VERSION}
                                  PUBLIC_HEADER
-                                 "${Opus_PUBLIC_HEADER}")
+                                 "${Opus_PUBLIC_HEADER}"
+                                 POSITION_INDEPENDENT_CODE
+                                 True)
 
 target_include_directories(
   opus
diff --git a/external/linphone-sdk/external/opus/celt/arm/celt_neon_intr.c b/external/linphone-sdk/external/opus/celt/arm/celt_neon_intr.c
index 69b90d7d..cd630a67 100644
--- a/external/linphone-sdk/external/opus/celt/arm/celt_neon_intr.c
+++ b/external/linphone-sdk/external/opus/celt/arm/celt_neon_intr.c
@@ -381,6 +381,9 @@ static void xcorr_kernel_neon_float(const float32_t *x, const float32_t *y,
    vst1q_f32(sum, SUMM);
 }
 
+/* Workaround missing declaration in PRESUME_NEON mode */
+extern opus_val32 celt_inner_prod_neon(const opus_val16 *x, const opus_val16 *y, int N);
+
 void celt_pitch_xcorr_float_neon(const opus_val16 *_x, const opus_val16 *_y,
                         opus_val32 *xcorr, int len, int max_pitch, int arch) {
    int i;
diff --git a/external/linphone-sdk/external/opus/cmake/config.h.cmake.in b/external/linphone-sdk/external/opus/cmake/config.h.cmake.in
index 5550842c..138b29aa 100644
--- a/external/linphone-sdk/external/opus/cmake/config.h.cmake.in
+++ b/external/linphone-sdk/external/opus/cmake/config.h.cmake.in
@@ -1 +1,6 @@
-#cmakedefine PACKAGE_VERSION "@PACKAGE_VERSION@"
\ No newline at end of file
+#ifndef CONFIG_H
+#define CONFIG_H
+
+#cmakedefine PACKAGE_VERSION "@PACKAGE_VERSION@"
+
+#endif /* CONFIG_H */
