diff --git a/external/linphone-sdk/external/mbedtls/3rdparty/everest/CMakeLists.txt b/external/linphone-sdk/external/mbedtls/3rdparty/everest/CMakeLists.txt
index ee7f0e4b9a..68d203a72f 100644
--- a/external/linphone-sdk/external/mbedtls/3rdparty/everest/CMakeLists.txt
+++ b/external/linphone-sdk/external/mbedtls/3rdparty/everest/CMakeLists.txt
@@ -1,6 +1,6 @@
 set(everest_target "${MBEDTLS_TARGET_PREFIX}everest")
 
-add_library(${everest_target}
+add_library(${everest_target} STATIC
   library/everest.c
   library/x25519.c
   library/Hacl_Curve25519_joined.c)
diff --git a/external/linphone-sdk/external/mbedtls/3rdparty/p256-m/CMakeLists.txt b/external/linphone-sdk/external/mbedtls/3rdparty/p256-m/CMakeLists.txt
index 0e6c6d868b..440c2f1399 100644
--- a/external/linphone-sdk/external/mbedtls/3rdparty/p256-m/CMakeLists.txt
+++ b/external/linphone-sdk/external/mbedtls/3rdparty/p256-m/CMakeLists.txt
@@ -1,6 +1,6 @@
 set(p256m_target ${MBEDTLS_TARGET_PREFIX}p256m)
 
-add_library(${p256m_target}
+add_library(${p256m_target} STATIC
     p256-m_driver_entrypoints.c
     p256-m/p256-m.c)
 
diff --git a/external/linphone-sdk/external/mbedtls/CMakeLists.txt b/external/linphone-sdk/external/mbedtls/CMakeLists.txt
index bc8ea2a341..924b40d246 100644
--- a/external/linphone-sdk/external/mbedtls/CMakeLists.txt
+++ b/external/linphone-sdk/external/mbedtls/CMakeLists.txt
@@ -20,6 +20,8 @@
 # until our infrastructure catches up.
 cmake_minimum_required(VERSION 3.5.1)
 
+set(MBEDTLS_VERSION "3.6")
+
 include(CMakePackageConfigHelpers)
 
 # Include convenience functions for printing properties and variables, like
@@ -37,6 +39,12 @@ cmake_policy(SET CMP0011 NEW)
 # is deprecated and will be removed in future versions.
 cmake_policy(SET CMP0012 NEW)
 
+# CMP0077 policy is required by Flexisip build. Remove it once the CMake required
+# version is higer or equal to 3.13.
+if(NOT CMAKE_VERSION VERSION_LESS 3.13)
+    cmake_policy(SET CMP0077 NEW)
+endif()
+
 if(TEST_CPP)
     project("Mbed TLS"
         LANGUAGES C CXX
@@ -315,6 +323,10 @@ if(LIB_INSTALL_DIR)
     set(CMAKE_INSTALL_LIBDIR "${LIB_INSTALL_DIR}")
 endif()
 
+if("${MBEDTLS_TARGET_PREFIX}" STREQUAL "")
+    set(MBEDTLS_FORCED_TARGET_PREFIX "mbedtls-")
+endif()
+
 if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/framework/CMakeLists.txt")
     if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
         message(FATAL_ERROR "${CMAKE_CURRENT_SOURCE_DIR}CMakeLists.txt not found (and does appear to be a git checkout). Run `git submodule update --init` from the source tree to fetch the submodule contents.")
@@ -450,7 +462,7 @@ if(ENABLE_PROGRAMS)
     add_subdirectory(programs)
 endif()
 
-ADD_CUSTOM_TARGET(${MBEDTLS_TARGET_PREFIX}apidoc
+ADD_CUSTOM_TARGET(${MBEDTLS_TARGET_PREFIX}${MBEDTLS_FORCED_TARGET_PREFIX}apidoc
     COMMAND doxygen mbedtls.doxyfile
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doxygen)
 
diff --git a/external/linphone-sdk/external/mbedtls/include/mbedtls/mbedtls_config.h b/external/linphone-sdk/external/mbedtls/include/mbedtls/mbedtls_config.h
index 8fa445a132..42dfde12c8 100644
--- a/external/linphone-sdk/external/mbedtls/include/mbedtls/mbedtls_config.h
+++ b/external/linphone-sdk/external/mbedtls/include/mbedtls/mbedtls_config.h
@@ -2052,7 +2052,7 @@
  *
  * Uncomment this to enable support for use_srtp extension.
  */
-//#define MBEDTLS_SSL_DTLS_SRTP
+#define MBEDTLS_SSL_DTLS_SRTP
 
 /**
  * \def MBEDTLS_SSL_DTLS_CLIENT_PORT_REUSE
@@ -2182,7 +2182,7 @@
  *
  * Uncomment this to allow your own alternate threading implementation.
  */
-//#define MBEDTLS_THREADING_ALT
+#define MBEDTLS_THREADING_ALT
 
 /**
  * \def MBEDTLS_THREADING_PTHREAD
@@ -3498,7 +3498,7 @@
  * Uncomment to have the library check for the Armv8-A SHA-256 crypto extensions
  * and use them if available.
  */
-//#define MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_IF_PRESENT
+#define MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_IF_PRESENT
 
 /**
  * \def MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT
@@ -3775,7 +3775,7 @@
  *
  * Enable this layer to allow use of mutexes within Mbed TLS
  */
-//#define MBEDTLS_THREADING_C
+#define MBEDTLS_THREADING_C
 
 /**
  * \def MBEDTLS_TIMING_C
diff --git a/external/linphone-sdk/external/mbedtls/include/mbedtls/threading.h b/external/linphone-sdk/external/mbedtls/include/mbedtls/threading.h
index 7b55a46496..b886cac176 100644
--- a/external/linphone-sdk/external/mbedtls/include/mbedtls/threading.h
+++ b/external/linphone-sdk/external/mbedtls/include/mbedtls/threading.h
@@ -41,7 +41,9 @@ typedef struct mbedtls_threading_mutex_t {
 
 #if defined(MBEDTLS_THREADING_ALT)
 /* You should define the mbedtls_threading_mutex_t type in your header */
-#include "threading_alt.h"
+//#include "threading_alt.h"
+// mbedtls_threading_mutex_t is defined as void * so it can be cast to any pointer
+typedef void* mbedtls_threading_mutex_t;
 
 /**
  * \brief           Set your alternate threading implementation function
diff --git a/external/linphone-sdk/external/mbedtls/library/CMakeLists.txt b/external/linphone-sdk/external/mbedtls/library/CMakeLists.txt
index 193593f753..41cd8b7ab9 100644
--- a/external/linphone-sdk/external/mbedtls/library/CMakeLists.txt
+++ b/external/linphone-sdk/external/mbedtls/library/CMakeLists.txt
@@ -205,10 +205,16 @@ endif()
 
 if(CMAKE_COMPILER_IS_GNUCC)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-declarations")
+    if(UNIX AND CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8-a+crypto")
+    endif()
 endif(CMAKE_COMPILER_IS_GNUCC)
 
 if(CMAKE_COMPILER_IS_CLANG)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-declarations -Wdocumentation -Wno-documentation-deprecated-sync -Wunreachable-code")
+    if(UNIX AND CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8-a+crypto")
+    endif()
 endif(CMAKE_COMPILER_IS_CLANG)
 
 if(CMAKE_COMPILER_IS_MSVC)
@@ -261,6 +267,7 @@ set(mbedx509_target   ${mbedx509_target}   PARENT_SCOPE)
 set(mbedcrypto_target ${mbedcrypto_target} PARENT_SCOPE)
 
 if (USE_STATIC_MBEDTLS_LIBRARY)
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
     set(mbedtls_static_target    ${mbedtls_target})
     set(mbedx509_static_target   ${mbedx509_target})
     set(mbedcrypto_static_target ${mbedcrypto_target})
@@ -284,8 +291,9 @@ set(everest_target "${MBEDTLS_TARGET_PREFIX}everest")
 
 if(USE_STATIC_MBEDTLS_LIBRARY)
     add_library(${mbedcrypto_static_target} STATIC ${src_crypto})
-    set_target_properties(${mbedcrypto_static_target} PROPERTIES OUTPUT_NAME mbedcrypto)
+    set_target_properties(${mbedcrypto_static_target} PROPERTIES OUTPUT_NAME mbedcrypto VERSION 3.6.6)
     target_link_libraries(${mbedcrypto_static_target} PUBLIC ${libs})
+    target_compile_definitions(${mbedcrypto_static_target} PUBLIC "MBEDTLS_STATIC")
 
     if(TARGET ${everest_target})
         target_link_libraries(${mbedcrypto_static_target} PUBLIC ${everest_target})
@@ -296,12 +304,14 @@ if(USE_STATIC_MBEDTLS_LIBRARY)
     endif()
 
     add_library(${mbedx509_static_target} STATIC ${src_x509})
-    set_target_properties(${mbedx509_static_target} PROPERTIES OUTPUT_NAME mbedx509)
+    set_target_properties(${mbedx509_static_target} PROPERTIES OUTPUT_NAME mbedx509 VERSION 3.6.6)
     target_link_libraries(${mbedx509_static_target} PUBLIC ${libs} ${mbedcrypto_static_target})
+    target_compile_definitions(${mbedx509_static_target} PUBLIC "MBEDTLS_STATIC")
 
     add_library(${mbedtls_static_target} STATIC ${src_tls})
-    set_target_properties(${mbedtls_static_target} PROPERTIES OUTPUT_NAME mbedtls)
+    set_target_properties(${mbedtls_static_target} PROPERTIES OUTPUT_NAME mbedtls VERSION 3.6.6)
     target_link_libraries(${mbedtls_static_target} PUBLIC ${libs} ${mbedx509_static_target})
+    target_compile_definitions(${mbedtls_static_target} PUBLIC "MBEDTLS_STATIC")
 
     if(GEN_FILES)
         add_dependencies(${mbedcrypto_static_target}
@@ -372,7 +382,9 @@ foreach(target IN LISTS target_libraries)
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
 endforeach(target)
 
-set(lib_target "${MBEDTLS_TARGET_PREFIX}lib")
+install(EXPORT MbedTLSTargets DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/mbedtls")
+
+set(lib_target "${MBEDTLS_TARGET_PREFIX}${MBEDTLS_FORCED_TARGET_PREFIX}lib")
 
 add_custom_target(${lib_target} DEPENDS ${mbedcrypto_target} ${mbedx509_target} ${mbedtls_target})
 if(USE_STATIC_MBEDTLS_LIBRARY AND USE_SHARED_MBEDTLS_LIBRARY)
