diff --git a/external/linphone-sdk/cmake/ExternalDependencies.cmake b/external/linphone-sdk/cmake/ExternalDependencies.cmake
index dbc2051fa5..1eea4b2cd4 100755
--- a/external/linphone-sdk/cmake/ExternalDependencies.cmake
+++ b/external/linphone-sdk/cmake/ExternalDependencies.cmake
@@ -942,7 +942,7 @@ if(BUILD_LIBXML2)
 		set(BUILD_SHARED_LIBS ${BUILD_LIBXML2_SHARED_LIBS})
 
 		add_subdirectory("external/libxml2")
-		add_dependencies(sdk xml2)
+		add_dependencies(sdk LibXml2)
 	endfunction()
 	add_xml2()
 endif()
